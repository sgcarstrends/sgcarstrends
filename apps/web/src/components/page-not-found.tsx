"use client";

import { Button } from "@heroui/react";
import Typography from "@web/components/typography";
import { ArrowLeft, Home } from "lucide-react";
import Link from "next/link";

export const PageNotFound = () => (
  <div className="container mx-auto px-4 py-16">
    <div className="flex min-h-[60vh] flex-col items-center justify-center text-center">
      <div className="mb-8">
        <Typography.Display className="mb-4 text-primary">
          404
        </Typography.Display>
        <Typography.H1 className="mb-2">Page Not Found</Typography.H1>
        <Typography.BodyLarge className="max-w-md text-foreground-500">
          The page you&apos;re looking for doesn&apos;t exist or has been moved.
        </Typography.BodyLarge>
      </div>

      <div className="mb-12 flex flex-col gap-4 sm:flex-row">
        <Button
          as={Link}
          href="/"
          color="primary"
          variant="solid"
          startContent={<Home className="size-4" />}
          size="lg"
        >
          Go to Homepage
        </Button>
        <Button
          variant="bordered"
          onPress={() => window.history.back()}
          startContent={<ArrowLeft className="size-4" />}
          size="lg"
        >
          Go Back
        </Button>
      </div>

      <Typography.BodySmall className="mt-8 text-foreground-500">
        Need help? Visit our{" "}
        <Link href="/faq" className="text-primary hover:underline">
          FAQ page
        </Link>{" "}
        or go back to the{" "}
        <Link href="/" className="text-primary hover:underline">
          homepage
        </Link>
        .
      </Typography.BodySmall>
    </div>
  </div>
);
