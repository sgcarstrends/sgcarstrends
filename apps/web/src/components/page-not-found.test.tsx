import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { PageNotFound } from "./page-not-found";

describe("PageNotFound", () => {
  it("renders the 404 error message", () => {
    const { container } = render(<PageNotFound />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText("404")).toBeInTheDocument();
    expect(screen.getByText("Page Not Found")).toBeInTheDocument();
    expect(
      screen.getByText(
        "The page you're looking for doesn't exist or has been moved.",
      ),
    ).toBeInTheDocument();
  });

  it("renders navigation buttons", () => {
    render(<PageNotFound />);

    expect(screen.getByText("Go to Homepage")).toBeInTheDocument();
    expect(screen.getByText("Go Back")).toBeInTheDocument();
  });

  it("renders help text with links", () => {
    render(<PageNotFound />);

    expect(screen.getByText(/Need help\? Visit our/)).toBeInTheDocument();
    expect(screen.getByText(/or go back to the/)).toBeInTheDocument();
  });

  it("has correct link href attributes", () => {
    render(<PageNotFound />);

    const faqLink = screen.getByRole("link", { name: /faq page/i });
    expect(faqLink).toHaveAttribute("href", "/faq");
  });
});
