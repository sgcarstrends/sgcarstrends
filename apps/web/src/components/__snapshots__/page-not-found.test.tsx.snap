// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNotFound > renders the 404 error message 1`] = `
<div>
  <div
    class="container mx-auto px-4 py-16"
  >
    <div
      class="flex min-h-[60vh] flex-col items-center justify-center text-center"
    >
      <div
        class="mb-8"
      >
        <h1
          class="scroll-m-20 font-semibold text-4xl tracking-tight lg:text-5xl w-fit bg-gradient-to-r from-foreground to-foreground/50 bg-clip-text leading-normal transition-opacity text-primary"
        >
          404
        </h1>
        <h2
          class="scroll-m-20 font-semibold text-3xl text-foreground tracking-tight first:mt-0"
        >
          Page Not Found
        </h2>
        <p
          class="text-foreground text-lg leading-relaxed"
        >
          The page you're looking for doesn't exist or has been moved.
        </p>
      </div>
      <div
        class="mb-12 flex flex-col gap-4 sm:flex-row"
      >
        <a
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-6 min-w-24 h-12 text-medium gap-3 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
          data-react-aria-pressable="true"
          href="/"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-house size-4"
            fill="none"
            focusable="false"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
            />
            <path
              d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
            />
          </svg>
          Go to Homepage
        </a>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium px-6 min-w-24 h-12 text-medium gap-3 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-transparent border-default text-foreground data-[hover=true]:opacity-hover"
          data-react-aria-pressable="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-arrow-left size-4"
            fill="none"
            focusable="false"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m12 19-7-7 7-7"
            />
            <path
              d="M19 12H5"
            />
          </svg>
          Go Back
        </button>
      </div>
      <p
        class="text-default-600 text-sm leading-6"
      >
        Need help? Visit our
         
        <a
          class="text-primary hover:underline"
          href="/faq"
        >
          FAQ page
        </a>
         
        or go back to the
         
        <a
          class="text-primary hover:underline"
          href="/"
        >
          homepage
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
